### PR TITLE
Make multi-type annotation settings match docs

### PR DESCRIPTION
**Pull request by treuherz**
The Docker docs in multiple places describe passing an annotation at the command line like "index,manifest:com.example.name=my-cool-image", and say that this will result in the annotation being applied to both the index and the manifest. It doesn't seem like this was actually implemented, and instead it just results in an annotation key with "index,manifest:" at the beginning being applied to the manifest.

This change splits the part of the key before the colon by comma, and creates an annotation for each type/platform given, so the implementation should now match the docs.

### CODE REVIEW

**Comment by treuherz** [comment]
I get the same test failures locally with and without this change - I think these must be my setup rather than an actual problem. Hopefully the validation will be green but I'll mark as draft again if not.

**Comment by treuherz** [comment]
A dep changed from `// indirect` to direct  - fixed, re-run vendor etc. Sorry about that

File: util/buildflags/export_test.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+					},
+					Key: "com.example",
+				}: "a",
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got, gotErr := ParseAnnotations(test.in)
```
Comment (tonistiigi) [inline util/buildflags/export_test.go:89]: `s/gotErr/err/`

File: util/buildflags/export_test.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+			if test.wantErr != "" {
+				require.ErrorContains(t, gotErr, test.wantErr)
+			} else {
+				assert.NoError(t, gotErr)
```
Comment (tonistiigi) [inline util/buildflags/export_test.go:93]: `NoError` should use `require`

File: util/buildflags/export.go [hunk 3]
```diff
@@ -81,7 +81,8 @@ func ParseExports(inp []string) ([]*controllerapi.ExportEntry, error) {
 
 func ParseAnnotations(inp []string) (map[exptypes.AnnotationKey]string, error) {
 	// TODO: use buildkit's annotation parser once it supports setting custom prefix and ":" separator
-	annotationRegexp := regexp.MustCompile(`^(?:([a-z-]+)(?:\[([A-Za-z0-9_/-]+)\])?:)?(\S+)$`)
+	annotationRegexp := regexp.MustCompile(`^((?:[a-z-]+(?:\[[A-Za-z0-9_/-]+\])?)(?:,[a-z-]+(?:\[[A-Za-z0-9_/-]+\])?)*:)?(\S+)$`)
```
Comment (tonistiigi) [inline util/buildflags/export.go:84]: This needs comments about why there are multiple and what's the difference

File: util/buildflags/export_test.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+			}
+			for k, v := range got {
+				gotKVs = append(gotKVs, kv{k, v})
+			}
+
+			sortFunc := func(a, b kv) int { return cmp.Compare(a.Key.String(), b.Key.String()) }
+			slices.SortFunc(wantKVs, sortFunc)
+			slices.SortFunc(gotKVs, sortFunc)
+
+			if diff := gocmp.Diff(wantKVs, gotKVs); diff != "" {
```
Comment (tonistiigi) [inline util/buildflags/export_test.go:113]: I think you can just use `assert.Equal` in here. No need for `gocmp`.

File: util/buildflags/export_test.go [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got, gotErr := ParseAnnotations(test.in)
+			if test.wantErr != "" {
+				require.ErrorContains(t, gotErr, test.wantErr)
+			} else {
+				assert.NoError(t, gotErr)
+			}
+
+			// Can't compare maps with pointer in their keys, need to extract and sort the map entries
```
Comment (tonistiigi) [inline util/buildflags/export_test.go:96]: might make sense to make this covert code into a function and then convert both maps. Atm there is duplication where exactly the same commands need to run on both maps.

File: util/buildflags/export.go (hunk 3 above)
Comment (treuherz) [reply util/buildflags/export.go:84]: The regexp is pretty inscrutable, I might try breaking it up into cuts and splits and just using a smaller regexp for pulling out the platform specifier, if it’s all the same to you

File: util/buildflags/export_test.go (hunk 5 above)
Comment (treuherz) [reply util/buildflags/export_test.go:95]: done

File: util/buildflags/export_test.go (hunk 4 above)
Comment (treuherz) [reply util/buildflags/export_test.go]: done

File: util/buildflags/export_test.go (hunk 1 above)
Comment (treuherz) [reply util/buildflags/export_test.go]: done

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**